process/project_milestones.md

```diff
@@ -0,0 +1,51 @@
+
+## Sandbox
+
+Some milestones may not apply to all projects and should be leveraged as guiding points. Milestones are described as the desired state with examples for how to achieve this.
+* Project has the basics of security underway
+  * Completion of the [self-assessment](https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md)
+* Project holds community meetings or active discussions
+  * Zoom with meeting notes occurring more than twice a quarter (or community.cncf.io!) 
+  * Use of GitHub discussions, issues, or a group messaging service (slack, wechat, etc.)
+* Project guides community contributions
+  * PRs from community members receive timely feedback and discussion from the maintainers
+* Project works with an angel adopter
+  * Partners closely with the project to address organizational needs of the project early on
+* Project begins solidifying a versioning schema and release cadence
+  * A roadmap clearly defines initiatives that align with releases
+  * Releases begin to have establish a pattern of regularity
+* Project has engaged a Technical Advisory Group for feedback on some portion of the project relevant to the domain
+* Project has a clearly discoverable governance doc that covers the basics of decision making to include the commit bit
+* TBD - add more!
+
+## Incubation
+
+Incubation milestones are guiding points to highlight success areas a maturing project can accomplish before Graduation. Not all Incubating projects will achieve every milestone, however many successful graduated projects have demonstrated these milestones prior to their Graduation. How each project achieves some or all of these milestones will vary widely. Projects do not need to meet this in order to graduate, rather these assist projects in meeting the Graduation criteria.
+* Project has a robust set of governance documentation that defines how the direction of the project is decided upon and managed which considers company diversity in decision making
+* Project can demonstrate its application, practice, and adjustments to its governance documentation as a result of regular project operations
+* Project has a robust and mature security posture as evidenced by
+  * Threat model
+  * [Joint review](https://github.com/cncf/tag-security/tree/main/assessments#components-of-the-security-review-package)
+  * First passing then Silver OpenSSF Best Practice Badge 
+* Project documentation includes getting started guides, operating/administration instruction, security call-outs, and other core elements necessary to ease adoption.
+* Project provides scalability benchmarks and sets expectations around scaling
```

Comment:
This is unclear to me. Do we have examples or resources we could provide here?

Reply:
This would be a new discipline (and should include performance with scalability) - there are a few projects that have done comparisons of performance between themselves and other projects and products.  This would drop the comparison portion and rather focus on the recommended architecture, a few deployment options (single cluster, multi-cluster, multi-cloud), how it scales (increasing nodes, increased traffic, load, or processes, expected completion time, etc.) It may also be as simple as expressing the CPU utilization of running the project depending on various configurations.  The goal would be to set expectations with potential users/adopters that may also be beneficial in understand if the project is performing optimally in their environment or if there are other factors contributing to performance or scaling degradation.

Reply:
As a maintainer, I believe it's good to also define when this is _not_ a requirement. For example, KEDA handles autoscaling but there is no aspect of scalability/performance benchmark as we're just building on top of KEDA with an additional operator.

You can always make the case that this is still required; but for some projects I believe it is less beneficial or the impact on a project would be too big.

Again for KEDA, we'd need to spin up big (expensive) clusters with sample workloads (1000+) to do benchmarking of something that is just built on top of Kubernetes' building blocks (HPA / metric servers)